File: Domain/Entities/Payment.cs
using Alti.Domain.Common;
using Alti.Domain.Enums;

namespace Alti.Domain.Entities;

public class Payment : BaseEntity
{
    internal Payment() { }

    public int BookingId { get; internal set; }
    public decimal Amount { get; internal set; }
    public PaymentStatus Status { get; internal set; } = PaymentStatus.Pending;
    public string? ExternalReference { get; internal set; }
    public string? PaymentMethod { get; internal set; }
    public DateTimeOffset? ProcessedAt { get; internal set; }
}

File: Frontend/Web/src/models/Payment.ts
export interface PaymentResponseDto {
  id: number;
  bookingId: number;
  amount: number;
  status: number;
  externalReference?: string;
  paymentMethod?: string;
  processedAt?: string;
}

export interface CreatePaymentDto {
  bookingId: number;
  amount: number;
  paymentMethod: string;
}

export const PaymentStatusLabel: Record<number, string> = {
  0: "Pendiente",
  1: "Aprobado",
  2: "Rechazado",
  3: "Reembolsado",
};